File: src/main/java/com/saybetter/domain/solution/ui/dto/SolutionRequest.java
package com.saybetter.domain.solution.ui.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;

public class SolutionRequest {

	private SolutionRequest() {
		throw new IllegalStateException("Utility class");
	}

	@Getter
	@AllArgsConstructor
	public static class CreateSolution {
		private String title;
		private Integer baselineSessionTime;
		private Double interCompleteThresh;
		private Integer interMaintainTerm;
		private Integer commOptTimes;
		private Integer commOptCnt;
		private List<String> symbols;
	}
}

File: core/core-api/src/main/kotlin/io/say/better/domain/solution/ui/dto/SolutionRequest.kt
package io.say.better.domain.solution.ui.dto

import io.say.better.storage.mysql.domain.constant.ProgressState

class SolutionRequest private constructor() {
    init {
        throw IllegalStateException("Utility class")
    }

    data class CreateSolution constructor(
        val learnerEmail: String,
        val nowState: ProgressState,
        val educationGoal: String,
        val description: String,
        val title: String,
        val commOptTimes: Int,
        val commOptCnt: Int,
    )

    data class StartSolution(
        val sessionOrder: Int,
        val solutionId: Long,
        val nowStep: String,
        val sessionGoal: String? = null,
        val sessionDesc: String? = null,
    )

    data class EndSolution(
        val solutionId: Long,
        val progressId: Long,
        val orderNum: Int,
        val createRecordSymbols: List<CreateRecordSymbol>,
        var reviewId: Long? = null,
    )

    data class CreateRecordSymbol(
        val symbolId: Long,
        val touchOrder: Int,
        val touchTime: String,
    )
}
